classes/converter.php

```diff
@@ -78,9 +78,10 @@ final public function convert_moodle_url_to_pdf(moodle_url $url, string $filenam
                                               string $cookiename = '', string $cookievalue = ''): \stored_file {
 
         try {
-            $key = helper::create_user_key();
-            $proxyurl = helper::get_proxy_url($url, $key);
             $filename = ($filename === '') ? helper::get_moodle_url_pdf_filename($url) : $filename;
+            $instanceid = key_manager::get_instance_id($filename);
```

Comment:
can we move this inside create_user_key? Basically we create a key for one URL only, right?  So knowing URL we can always recalculate instanceid. 

So basically we should rename our method key_manager::create_user_key to key_manager::create_user_key_for_url(string $url)

then inside we can do our encoding to save it to DB. 

 then  login_with_key  should get key and url to check if it's ok to use this key for that url like login_with_key(string $key, int $url) and then key_manager can check that.

In this case it's much more clear that we use key for url and not for abstract instanceid

Reply:
Good idea, I'll implement this

Reply:
Abstracted the instance generation and instead use the URL in `key_manager` and `login_manager` function parameters